_resources/port1.0/group/cmake-1.1.tcl

```diff
@@ -284,14 +471,31 @@ platform darwin {
 configure.universal_args-delete --disable-dependency-tracking
 
 variant debug description "Enable debug binaries" {
-    configure.pre_args-replace  -DCMAKE_BUILD_TYPE=MacPorts \
-                                -DCMAKE_BUILD_TYPE=Debug
+    # this PortGroup uses a custom CMAKE_BUILD_TYPE giving complete control over
+    # the compiler flags. We use that here: replace the default -O2 with -O0, add
+    # debugging options and do otherwise an exactly identical build.
+    configure.cflags-replace         -O2 -O0
+    configure.cxxflags-replace       -O2 -O0
+    configure.objcflags-replace      -O2 -O0
+    configure.objcxxflags-replace    -O2 -O0
+    configure.ldflags-replace        -O2 -O0
+    # get most if not all possible debug info
+    if {[string match *clang* ${configure.cxx}] || [string match *clang* ${configure.cc}]} {
+        set cmake::debugopts "-g -fno-limit-debug-info -DDEBUG"
+    } else {
+        set cmake::debugopts "-g -DDEBUG"
+    }
+    configure.cflags-append         ${cmake::debugopts}
+    configure.cxxflags-append       ${cmake::debugopts}
+    configure.objcflags-append      ${cmake::debugopts}
+    configure.objcxxflags-append    ${cmake::debugopts}
+    configure.ldflags-append        ${cmake::debugopts}
+    # try to ensure that info won't get stripped
+    configure.args-append           -DCMAKE_STRIP:FILEPATH=/bin/echo
 }
 
-default build.dir {${configure.dir}}
-
-default build.post_args {VERBOSE=ON}
-
-# Generated Unix Makefiles contain a "fast" install target that begins
-# installing immediately instead of checking build dependencies again.
-default destroot.target {install/fast}
+# cmake doesn't like --enable-debug, so in case a portfile sets
+# --enable-debug (regardless of variant) we remove it
+if {[string first "--enable-debug" ${configure.args}] > -1} {
+    configure.args-delete     --enable-debug
+}
```

Comment:
The same statement already exists [above](https://github.com/macports/macports-ports/pull/1028/commits/00edba123fb5db4d278457152dd4f9acf9d7fa42#diff-1cfa208ef1d2a7a0f9ffef790a8c8088R350).

I even assume this is useless here, as it is evaluated before the rest of the Portfile. And even then, why is this necessary? Just do not add `--enable-debug` in the Portfile? I am confused.

Reply:
You're right, I must have intended to move the code and forgot to remove it from the original location. As far as I can tell I inherited this morsel with the rest of cmake-1.0.tcl and I guess that in turn it may have been implemented because of the debug 1.0 PG . 

We could remove it, or rather I should probably undo the effect of that PG's current actions (setting the build type to debugFull while we already use a build type that makes cmake honour `configure.cflags` and `configure.cxxflags`).